test/RelationTest.php

```diff
@@ -133,6 +170,26 @@ public function testWhereHash()
         $this->assertEquals('Bill Clinton', $authors[0]->name);
     }
 
+    public function testSelect()
+    {
+        $relation = Author::select('name')->where("mixedCaseField = 'Bill'")->last;
+        $this->assertEquals('Uncle Bob', $relation->name);
+
+        $this->expectException(UndefinedPropertyException::class);
+        $columnNotInSelectList = $relation->parent_author_id;
+    }
+
+    public function testSelectChainedSelects()
+    {
+        $relation = Author::select('name,name')->select('author_id');
```

Comment:
Looking at this test, it's not clear to me what the bug is you're fixing. I'm not saying you're wrong, just that it's not clear. For starters, using reflection to dip into internals as proof that there is a problem is sort of breaking the rules. If you want to show a problem, do it with publicly available means. For instance:

```php
$author = Author::select('name,name')->take();
$this->assertEquals(['name' => 'Tito'], $author->attributes() );
```

This is why I suggested starting with a failing test case for us to look at before diving in. Recommend starting a new branch that has a single failing test case, preferably performed without the kind of introspection you're doing here. Once we understand the situation better, we can talk about what the fix is.

Reply:
Hopefully the new commit explains things better. It prevents:

SELECT name, name
FROM Author

and

SELECT name, *
FROM Author

Reply:
Okie-doke, then 👍